test/Collector.test.ts

```diff
@@ -0,0 +1,185 @@
+import { expect, use } from 'chai';
+import Collector from '../build/contracts/Collector.json';
+import TestToken from '../build/contracts/TestToken.json';
+import {
+    deployContract,
+    loadFixture,
+    MockProvider,
+    solidity
+} from 'ethereum-waffle';
+import { BaseContract } from '@ethersproject/contracts';
+import { Wallet } from '@ethersproject/wallet';
+
+use(solidity);
+
+const buildPartners = (
+    wallets: Wallet[],
+    share1?: number,
+    share2?: number,
+    share3?: number,
+    share4?: number
+) => {
+    const [owner, remainder, partner1, partner2, partner3, partner4] = wallets;
+
+    return [
+        {
+            beneficiary: partner1.getAddress(),
+            share: share1 ?? 20
+        },
+        {
+            beneficiary: partner2.getAddress(),
+            share: share2 ?? 35
+        },
+        {
+            beneficiary: partner3.getAddress(),
+            share: share3 ?? 13
+        },
+        {
+            beneficiary: partner4.getAddress(),
+            share: share4 ?? 32
+        }
+    ];
+};
+
+async function deployCollector(
+    wallets: Wallet[],
+    testToken: BaseContract,
+    partners: any
+) {
+    const [ownerWallet, remainderWallet] = wallets;
+
+    const collector = await deployContract(ownerWallet, Collector, [
+        ownerWallet.address,
+        testToken.address,
+        partners,
+        remainderWallet.address
+    ]);
+
+    return { collector };
+}
+describe('Collector', function () {
+    async function deployTokenFixture(wallets) {
```

Comment:
Is it possible to type the wallets parameter?

Reply:
yes, done.